components/client/src/test/unit/java/com/hotels/styx/client/loadbalancing/strategies/BusyConnectionsStrategyStressTest.java

```diff
@@ -65,153 +51,153 @@ public void setUp() {
-        System.out.println("       new connections: " + metrics.counter("winner.newConnections").getCount());
-
-        double maxCount = results.values().stream().max(naturalOrder()).get();
-        double minCount = results.values().stream().min(naturalOrder()).get();
-
-        assertThat(results.size(), is(9));
-
-        assertThat(maxCount / minCount, is(lessThan(1.1)));
-    }
-
-    private Map<Origin, Integer> simulateLoadBalancerTraffic(List<SimulatedOrigin> origins, SimulatedApp app) {
-        Map<Origin, Integer> results = new TreeMap<>();
-
-        int totalTimeSec = 60 * 60;
-        int ticks = totalTimeSec * 1000;
-        Random random = new Random();
-
-        for (int i = 0; i < ticks; i++) {
-
-            boolean vote = app.newRequest(1000, random);
-
-            if (vote) {
-                Iterable<RemoteHost> pools = origins.stream().map(so ->
-                        remoteHost(so.origin, new StubConnectionPool(so.origin())
-                                .withBusyConnections(so.busyConnections())
-                                .withAvailableConnections(so.availableConnections()), mock(StyxHostHttpClient.class)))
-                        .collect(toList());
-
-                final BusyConnectionsStrategy strategy = new BusyConnectionsStrategy(() -> pools);
-
-                Iterable<RemoteHost> result = strategy.vote(contextFromSimulatedOrigins(origins));
-
-                RemoteHost winner = Iterables.get(result, 0);
-                if (winner.connectionPool().stats().availableConnectionCount() > 0) {
-                    metrics.counter("winner.existingConnections").inc();
-                } else {
-                    metrics.counter("winner.newConnections").inc();
-                }
-
-                results.putIfAbsent(winner.connectionPool().getOrigin(), 0);
-                results.computeIfPresent(winner.connectionPool().getOrigin(), (k, v) -> ++v);
-
-                app.borrowFor(i, winner);
-            }
-
-            for (SimulatedOrigin so : origins) {
-                so.advanceTime(i, 1000, random);
-            }
-        }
-
-        return results;
-    }
-
-    private LoadBalancingStrategy.Context contextFromSimulatedOrigins(List<SimulatedOrigin> origins) {
-        return new LoadBalancingStrategy.Context() {
-            @Override
-            public Id appId() {
-                return null;
-            }
-
-            @Override
-            public HttpRequest currentRequest() {
-                return null;
-            }
-
-            @Override
-            public double oneMinuteRateForStatusCode5xx(Origin origin) {
-                SimulatedOrigin result = origins.stream()
-                        .filter(so -> so.origin().equals(origin))
-                        .findFirst()
-                        .get();
-
-                return result.status500Rate();
-            }
-        };
-    }
+//    @Test(dataProvider = "allOrigins")
```

Comment:
Commented out code

Reply:
Do you guys think we still need this StressTest? What is it supposed to demonstrate? That all origins get some traffic? Our performance pipeline should be sufficient for that purpose. What do you think?